scenarios.md

```diff
@@ -0,0 +1,190 @@
+# Notary Signing - Scenarios
+
+As containers and cloud native artifacts become the common unit of deployment, users want to know the artifacts in their private registries and the artifacts deployed are the same artifacts that were initially published.
+
+The [OCI TOB][oci-tob] has adopted [OCI Artifacts][artifacts-repo], generalizing container images as one of many types of artifacts that may be stored in a registry. Other artifact types currently include:
+
+* [Helm Charts][helm-registry]
+* [Singularity][singularity]
+* Car firmware updates, deployed from OCI Artifact registries
+
+## The Need for a Generalized Signing Solution
+
+This document serves as the requirements and constraints of a generalized signing solution. It focuses on the scenarios and needs, and very specifically avoids any reference to other projects or implementations. As our working group forms a consensus on the requirements, the group will then transition to a spec.
+
+
+## Key Stake Holders & Contributors
+
+As we identify the requirements and constraints, a number of key contributors will be asked to represent their requirements and constraints.
+
+> Please add companies, projects, products that you believe should be included.
+
+* Registry Cloud Operators
+  * [Azure Container Registry (acr)][acr] - Steve Lasker <steve.lasker@microsoft.com> (@stevelasker)
+  * [AWS Container Registry (ecr)][ecr] - Omar Paul <omarpaul@amazon.com>
+  * [Docker Hub][docker-hub]
+  * [Google Container Registry (gcr)][gcr]
+  * [GitHub Package Registry (gpr)][gpr]
+  * [Quay][quay]
+  * [IBM Cloud Container Registry (icr)][icr]
+* Registry Vendors, Projects & Products
+  * [Docker Trusted Registry][docker-dtr]
+  * [Harbor][harbor]
+  * [JFrog Artifactory][jfrog]
+* Artifact Types
+  * Container Images - OCI Org
+  * Helm Charts
+  * Singularity
+  * Operator Bundles
+
+## Scenarios
+
+### Scenario #1: Local Build, Sign, Validate
+
+Prior to doing any deployment, a developer can test the: build, sign, validate scenario.
+
+1. Locally build a container image using a non-registry specific `name:tag`, such as:  
```

Comment:
ugh, the dreaded default registry. Ok. I've added the following:
- The lack of a registry name does not infer docker.io as the default registry. This does imply changes to either the docker client, the notary client or both.

@dmcgowan do you have any work in flight with containerd that addresses the default registry issues?

Reply:
Not sure what you mean by "the default registry issue". If you try to push an image called `hello-world:1` with the containerd client (or `ctr`) it will spit it back out at you for not defining a registry. If you put `hello-world:1` into Docker or Kubernetes (CRI), it will interpret it as `docker.io/library/hello-world:1`. This won't ever change in Docker and likely never will in Kubernetes (CRI) either since it would just break everyone for no real usability gain. From a notary perspective, it should only ever see `docker.io/library/hello-world:1` and leave it to the clients to take care of interpreting short names according to their own UX experience. I wouldn't really categorize it as the dreaded default registry though, each client is free to make their own opinionated UX. Notary should have no opinions at its core, in its tool, or in its documentation. I would suggest just using examples which include a full registry name except where you are explicitly giving an example of an official image in the Docker UX.

Reply:
> Default registry issue:

The docker client assumption that all un-scoped images belong to docker hub (docker.io)

> If you try to push an image called hello-world:1 with the containerd client (or ctr) it will spit it back out at you for not defining a registry. 

This is the desired behavior as it's presumptive for an open source project to default to a specific vendor. This has caused lots of issues and frustrations. 

> From a notary perspective, it should only ever see docker.io/library/hello-world:1 and leave it to the clients to take care of interpreting short names 

Agreed. As we venture into Notary v2, and new tooling, I'm asking we don't assume a default registry, nor assume one must be assigned. If a user names an entity `foo:1`, it's really a loose artifact, and has no affinity to a registry until given a domain. Which could be `docker.io/someone/foo:1` or  `mydomain.com/foo:1`
The disconnect is Artifacts shouldn't be tied to a registry, rather they happen to be stored in them at various states. But, there's no home, or even assumption it must be assigned a registry. 
So, we just need to agree that an Artifact that doesn't have a domain, isn't associated with any registry. Which isn't an error, it's just a state.

Reply:
> So, we just need to agree that an Artifact that doesn't have a domain, isn't associated with any registry. Which isn't an error, it's just a state.

I think we should be careful here though. I have come to the conclusion that partial names are actually worse than no name at all. For example, in the OCI index specification it tried to define the reference name as only a tag component of a name (such as say "v1.0.0"), however, this made these images unable to be imported by clients without specifying the rest of the name and making a decision (which allows for opinionated interpretation) of how the full name would get reconstructed. Without a name, it is clear that a name should be provided and it can be anything. With a full name, also clear with assigning a new name a common and understand mechanism. 

I would suggest either recommending full names which include the registry or using a label which represents the "partial name". Just avoid having an ambiguous field that requires the client to figure out how to interpret the string, even just identifying domain/no domain isn't always straight forward to implement.

Reply:
> So, we just need to agree that an Artifact that doesn't have a domain, isn't associated with any registry. Which isn't an error, it's just a state.

I think a better interpretation here is really just that the signature is independent of the image name.  We already have that implication in scenario 4.

For this specific scenario, I'd suggest changing the wording to 

> Locally build a container image with a name and tag

and avoiding mention of the registry.  As long as the example doesn't include pushing the image, it's still clear that the image must be able to be signed and validated independently of its presence in a registry.

Reply:
please review the updated PR.